src/main/java/org/zalando/fahrschein/CursorManager.java

```diff
@@ -41,12 +45,14 @@ default void fromOldestAvailableOffset(String eventName, List<Partition> partiti
     }
 
     default void updatePartitions(String eventName, List<Partition> partitions) throws IOException {
+
         final Map<String, Cursor> cursorsByPartition = getCursors(eventName).stream().collect(Collectors.toMap(Cursor::getPartition, c -> c));
 
         for (Partition partition : partitions) {
+            final String minAvailableOffset = OFFSET_ORDERING.min(partition.getOldestAvailableOffset(), partition.getNewestAvailableOffset());
             final Cursor cursor = cursorsByPartition.get(partition.getPartition());
-            if (cursor == null || !partition.isAvailable(cursor.getOffset())) {
-                onSuccess(eventName, new Cursor(partition.getPartition(), partition.getOldestAvailableOffset()));
+            if (cursor == null || OFFSET_ORDERING.compare(cursor.getOffset(), minAvailableOffset) < 0) {
```

Comment:
there might be the use case to go for the `maxAvailableOffset` (different issue then).